src/main/g8/src/main/scala/worker/Main.scala

```diff
@@ -1,87 +1,95 @@
 import akka.pattern.ask
-import akka.actor.Identify
-import akka.actor.ActorIdentity
+import akka.persistence.journal.leveldb.{SharedLeveldbJournal, SharedLeveldbStore}
+import akka.util.Timeout
+import com.typesafe.config.{Config, ConfigFactory}
+
+import scala.concurrent.duration._
 
 object Main {
 
+  val backEndPortRange = 2000 to 2999
+  val frontEndPortRange = 3000 to 3999
+
   def main(args: Array[String]): Unit = {
-    if (args.isEmpty) {
-      startBackend(2551, "backend")
-      Thread.sleep(5000)
-      startBackend(2552, "backend")
-      startWorker(0)
-      Thread.sleep(5000)
-      startFrontend(0)
-    } else {
-      val port = args(0).toInt
-      if (2000 <= port && port <= 2999)
-        startBackend(port, "backend")
-      else if (3000 <= port && port <= 3999)
-        startFrontend(port)
-      else
-        startWorker(port)
+    args.headOption.map(_.toInt) match {
+      case None => startClusterInSameJvm()
+      case Some(port) if backEndPortRange.contains(port) => startBackEnd(port)
+      case Some(port) if frontEndPortRange.contains(port) => startFrontEnd(port)
+      case Some(port) => startWorker(port, args.lift(1).map(_.toInt).getOrElse(1))
     }
-
   }
 
-  def workTimeout = 10.seconds
-
-  def startBackend(port: Int, role: String): Unit = {
-    val conf = ConfigFactory.parseString(s"akka.cluster.roles=[$role]").
-      withFallback(ConfigFactory.parseString("akka.remote.netty.tcp.port=" + port)).
-      withFallback(ConfigFactory.load())
-    val system = ActorSystem("ClusterSystem", conf)
-
-    startupSharedJournal(system, startStore = (port == 2551), path =
-      ActorPath.fromString("akka.tcp://ClusterSystem@127.0.0.1:2551/user/store"))
-
-    system.actorOf(
-      ClusterSingletonManager.props(
-        Master.props(workTimeout),
-        PoisonPill,
-        ClusterSingletonManagerSettings(system).withRole(role)
-      ),
-      "master")
-
+  def startClusterInSameJvm(): Unit = {
+    // two backend nodes
+    startBackEnd(2551)
+    startBackEnd(2552)
+    // two front-end nodes
+    startFrontEnd(3000)
+    startFrontEnd(3001)
+    // two worker nodes with two worker actors each
+    startWorker(5001, 2)
+    startWorker(5002, 2)
   }
 
-  def startFrontend(port: Int): Unit = {
-    val conf = ConfigFactory.parseString("akka.remote.netty.tcp.port=" + port).
-      withFallback(ConfigFactory.load())
-    val system = ActorSystem("ClusterSystem", conf)
-    val frontend = system.actorOf(Props[Frontend], "frontend")
-    system.actorOf(Props(classOf[WorkProducer], frontend), "producer")
-    system.actorOf(Props[WorkResultConsumer], "consumer")
+  /**
+   * Start a node with the role backend on the given port. (This may also
+   * start the shared journal, see below for details)
+   */
+  def startBackEnd(port: Int): Unit = {
+    val system = ActorSystem("ClusterSystem", config(port, "back-end"))
+    startupSharedJournal(
```

Comment:
Hmm. I agree about the `SharedJournal` being an annoyance. "By the way, in this sample you cannot kill the master node because...". 

Could we even launch a cassandra instance from the demo main, isn't that done in the cassandra journal tests somehow?